chain/default_syncer.go

```diff
@@ -350,6 +348,11 @@ func (syncer *DefaultSyncer) HandleNewBlocks(ctx context.Context, blkCids []cid.
 	}
 	parent := parentTsas.TipSet
 
+	// Start locking now, so simultaneous calls to this method don't confuse each other about
+	// the state of the chain head.
+	syncer.mu.Lock()
+	defer syncer.mu.Unlock()
```

Comment:
We've moved this lock down so chain syncing won't deadlock with the longer timeout. We've analyzed the code and believe this is safe.

Reply:
Already talked to @acruikshank about this but acking that duplicate network traffic and duplicate block processing will happen as is and that this should change for efficiency.

Reply:
this looks like a really bad tradeoff to me, but ok. ac, i guess i ask that you explicitly call out the two consequences above of lock placement in a comment. also still curious about 1.

Reply:
@phritz I discussed this with @ZenGround0 and we went through root causes, potential DOS vectors and what happens when the syncer encounters valid forks. In the end we decided it was better to move the lock back where it was. We don't see any real consequences of stacking HandleNewBlock (now HandleNewTipset) calls pending a long chain sync, and it seems safer. If you identify problems that the longer timeout will cause with this locking, we should probably get on a call to discuss.